src/test/java/org/opendatakit/common/persistence/NewQueryResultTest.java

```diff
@@ -0,0 +1,231 @@
+/**
+ * Copyright (C) 2011 University of Washington
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.opendatakit.common.persistence;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.opendatakit.common.persistence.Query.Direction.ASCENDING;
+import static org.opendatakit.common.persistence.TestTable.booleanField;
+import static org.opendatakit.common.persistence.TestTable.dateField;
+import static org.opendatakit.common.persistence.TestTable.doubleField;
+import static org.opendatakit.common.persistence.TestTable.integerField;
+import static org.opendatakit.common.persistence.TestTable.stringField;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.opendatakit.common.persistence.exception.ODKDatastoreException;
+import org.opendatakit.common.persistence.exception.ODKEntityPersistException;
+import org.opendatakit.common.persistence.exception.ODKOverQuotaException;
+import org.opendatakit.common.security.User;
+import org.opendatakit.common.web.CallingContext;
+import org.opendatakit.common.web.TestContextFactory;
+
+
+public class NewQueryResultTest {
+  private static final int SET_SIZE = 3;

```

Comment:
Move this down, and set the value from the length of one of the arrays?

Reply:
Down where? do you mean the order of members is not right? Do we have a convention for that?

Regarding deriving the `SET_SIZE` value from the other array's length, I don't want to couple those things... It's just a coincidence that `SET_SIZE` is `3` and the arrays have 3 items...